helpers/osinfo.go

```diff
@@ -0,0 +1,270 @@
+package helpers
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+type OSReleaseID uint32
+
+func (o OSReleaseID) String() string {
+	return OSReleaseIDToString[o]
+}
+
+const (
+	UBUNTU OSReleaseID = iota + 1
+	FEDORA
+	ARCH
+	DEBIAN
+	CENTOS
+	STREAM
+	ALMA
+)
+
+// StringToOSReleaseID is a map of supported distributions
+var StringToOSReleaseID = map[string]OSReleaseID{
+	"ubuntu": UBUNTU,
+	"fedora": FEDORA,
+	"arch":   ARCH,
+	"debian": DEBIAN,
+	"centos": CENTOS,
+	"stream": STREAM,
+	"alma":   ALMA,
+}
+
+// OSReleaseIDToString is a map of supported distributions
+var OSReleaseIDToString = map[OSReleaseID]string{
+	UBUNTU: "ubuntu",
+	FEDORA: "fedora",
+	ARCH:   "arch",
+	DEBIAN: "debian",
+	CENTOS: "centos",
+	STREAM: "stream",
+	ALMA:   "alma",
+}
+
+const (
+	OS_NAME OSReleaseField = iota + 0
+	OS_ID
+	OS_ID_LIKE
+	OS_PRETTY_NAME
+	OS_VARIANT
+	OS_VARIANT_ID
+	OS_VERSION
+	OS_VERSION_ID
+	OS_VERSION_CODENAME
+	OS_BUILD_ID
+	OS_IMAGE_ID
+	OS_IMAGE_VERSION
+	OS_KERNEL_RELEASE // not part of default os-release, but we can use it here to facilitate things
+)
+
+type OSReleaseField uint32
+
+func (o OSReleaseField) String() string {
+	return OSReleaseFieldToString[o]
+}
+
+// StringToOSReleaseField is a map of os-release file fields
+var StringToOSReleaseField = map[string]OSReleaseField{
```

Comment:
please reconsider what needs to be exported and what not (across the PR)

Reply:
Those are all the fields os-release implementation supports. Not something I picked to support, if that is what you meant here. The API supports what https://www.freedesktop.org/software/systemd/man/os-release.html describes as fields in that file.

Reply:
I would agree that if we have the String() method, we don't need to export the map.

Reply:
Oh I see now.